asb-native/src/main/java/org/ballerinax/asb/receiver/MessageReceiver.java

```diff
@@ -18,340 +18,460 @@
      *
      * @param connectionString Azure service bus connection string.
-     * @param entityPath       Entity path (QueueName or SubscriptionPath).
+     * @param queueName        QueueName
+     * @param topicName        Topic Name
+     * @param subscriptionName Subscription Name
      * @param receiveMode      Receive Mode as PeekLock or Receive&Delete.
-     * @throws ServiceBusException  on failure initiating IMessage Receiver in Azure Service Bus instance.
-     * @throws InterruptedException on failure initiating IMessage Receiver due to thread interruption.
+     * @throws ServiceBusException  on failure initiating IMessage Receiver in Azure
+     *                              Service Bus instance.
+     * @throws InterruptedException on failure initiating IMessage Receiver due to
+     *                              thread interruption.
      */
-    public MessageReceiver(String connectionString, String entityPath, String receiveMode) throws ServiceBusException, InterruptedException {
-        this.entityPath = entityPath;
-        if (Objects.equals(receiveMode, RECEIVEANDDELETE)) {
-            this.receiver = ClientFactory.createMessageReceiverFromConnectionStringBuilder(
-                    new ConnectionStringBuilder(connectionString, entityPath), ReceiveMode.RECEIVEANDDELETE);
-        } else {
-            this.receiver = ClientFactory.createMessageReceiverFromConnectionStringBuilder(
-                    new ConnectionStringBuilder(connectionString, entityPath), ReceiveMode.PEEKLOCK);
+    public MessageReceiver(String connectionString, String queueName, String topicName, String subscriptionName,
+            String receiveMode)
+            throws ServiceBusException, InterruptedException {
+        ServiceBusReceiverClientBuilder receiverClientBuilder = new ServiceBusClientBuilder()
+                .connectionString(connectionString)
+                .receiver();
+        if (!queueName.isEmpty()) {
+            if (Objects.equals(receiveMode, RECEIVE_AND_DELETE)) {
+                this.receiver = receiverClientBuilder
+                        .receiveMode(ServiceBusReceiveMode.RECEIVE_AND_DELETE)
+                        .queueName(queueName)
+                        .buildClient();
+
+            } else {
+                this.receiver = receiverClientBuilder
+                        .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+                        .queueName(queueName)
+                        .buildClient();
+            }
+        } else if (!subscriptionName.isEmpty() && !topicName.isEmpty()) {
+            if (Objects.equals(receiveMode, RECEIVE_AND_DELETE)) {
+                this.receiver = receiverClientBuilder
+                        .receiveMode(ServiceBusReceiveMode.RECEIVE_AND_DELETE)
+                        .topicName(topicName)
+                        .subscriptionName(subscriptionName)
+                        .buildClient();
+
+            } else {
```

Comment:
Better tell somewhere in bal docs, PEEK_LOCK mode is the default.

Reply:
https://github.com/SanduDS/module-ballerinax-azure-service-bus/blob/cddb330359d18591b09b5c4f9c619e455840dee8/asb-ballerina/messageReceiver.bal#L40

already added here